dmoj/cptbox/helper.cpp

```diff
@@ -141,9 +199,25 @@ int cptbox_child_run(const struct child_config *config) {
     execve(config->file, config->argv, config->envp);
     perror("execve");
     return PTBOX_SPAWN_FAIL_EXECVE;
+}
 
-seccomp_fail:
-    return PTBOX_SPAWN_FAIL_SECCOMP;
+int has_landlock_check() {
+#if !PTBOX_FREEBSD
+    int landlock_version = landlock_create_ruleset(NULL, 0, LANDLOCK_CREATE_RULESET_VERSION);
+    if(landlock_version < 0) {
+        if(errno == ENOSYS || errno == EOPNOTSUPP) {
+            return 0;
+        }
+        else {
+            return -1;
+        }
+    }
+    else {
+        return 1;
+    }
```

Comment:
I hate this control flow.

```suggestion
    int landlock_version = landlock_create_ruleset(NULL, 0, LANDLOCK_CREATE_RULESET_VERSION);
    if (landlock_version >= 0)
        return 1;
    if (errno == ENOSYS || errno == EOPNOTSUPP)
        return 0;
    return -1;
```